### PR TITLE
fix(pre-launch science review): changing config, default behaviour fo…

### PR DESCRIPTION
**Pull request by franluca**
changing config, default behavior for aggregation strategy, introducing unknown ratio measure, splitting also considers \n

BREAKING CHANGES: ClassificationAccuracyConfig changing fields, introducing 2 more fields and renaming one

*Description of changes:* 
- (minor) Renamed column name for model output after converter function. reason: previous name was non-standard/misleading
- (minor) added splitting wrt \n with `re.split` for default converter function. reason: some models do not to put white spaces before/after new line, causing parsing to miss many valid outputs
- (major) reworked aggregation strategies for precision and recall. reason: previous default behavior was non-standard and did not consider differences of binary vs multiclass setting. 
   -  changes to ClassificationAccuracyConfig. To implement correct behaviour, the cleanest way is to change config. I added `binary_average_strategy` and `positive_label` to handle binary classification case. I changed the default of multiclass_average_strategy as `micro` is non-standard.
   - precision and recall are now computed only on the subset of records that are not classified as `unknown` 
- (major) added a further metrics that computes the ratio of unknown labels to compliment new behavior of precision and recall computation. It is fine if this measure is not shown in the report for now. We can follow up on this
- (minor) changed some docstrings 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fmeval/eval_algorithms/classification_accuracy.py [hunk 1]
```diff
@@ -46,13 +46,27 @@
...
 PRECISION_SCORE = "precision_score"
 RECALL_SCORE = "recall_score"
+UNKNOWN_RATIO = "unknown_ratio"
 UNKNOWN_LABEL = "unknown"
 PROMPT_COLUMN_NAME = "prompt"
-CLASSIFIED_MODEL_OUTPUT_COLUMN_NAME = "classified_model_output"
+PREDICTED_CLASS_COLUMN_NAME = "predicted_class"
+
+
+def unknown_ratio(y_pred) -> float:
```
[the remark this answers is not in the document]
Comment (danielezhu) [reply src/fmeval/eval_algorithms/classification_accuracy.py:55]: I think type annotations are tricky in these kinds of use cases where the argument is best described as a "1D array-like" (see [sklearn docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html) for example).

I supposed we could do something like `Union[pandas.DataFrame, numpy.array, List[Any]`, but that would be quite clunky, and not even necessarily exhaustive.

Comment (danielezhu) [inline src/fmeval/eval_algorithms/classification_accuracy.py:55]: Perhaps rename this to `unknown_fraction` instead?

File: src/fmeval/eval_algorithms/classification_accuracy.py [hunk 2]
```diff
@@ -88,15 +102,24 @@ class ClassificationAccuracyConfig(EvalAlgorithmConfig):
...
+        This determines how scores are aggregated in multiclass classification settings
         (see https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html).
-        Options are {'micro', 'macro', 'samples', 'weighted', 'binary'} or None, default='micro'.
+        Options are {'micro', 'macro', 'samples', 'weighted'} or None, default='weighted'.
     """
 
     valid_labels: Optional[List[str]] = None
     converter_fn: Callable[[str, List[str]], str] = convert_model_output_to_label
-    multiclass_average_strategy: Optional[str] = "micro"
+    binary_average_strategy: Optional[str] = "binary"
```
[the remark this answers is not in the document]
Comment (danielezhu) [reply src/fmeval/eval_algorithms/classification_accuracy.py:120]: I very much agree with Pola; "binary" is pretty much the only option that makes sense here. I suppose giving the user freedom to choose other options is fine, since we've configured the default appropriately. If they choose something like "micro" and are confused why precision and recall are always the same, that's a self-inflicted customer error.

File: src/fmeval/eval_algorithms/classification_accuracy.py [hunk 3]
```diff
@@ -274,14 +297,28 @@ def _generate_columns(row: Dict[str, Any]) -> Dict[str, Any]:  # pragma: no cove
 
     def _get_score(self, y_true, y_pred, eval_fn: Callable[..., float]) -> float:
```
Comment (danielezhu) [inline src/fmeval/eval_algorithms/classification_accuracy.py:298]: Please add unit tests for `_get_score` for the various cases (binary recall, binary precision, multiclass recall, multiclass precision, etc).

File: test/unit/eval_algorithms/test_classification_accuracy.py [hunk 4]
```diff
@@ -77,6 +78,15 @@
     EvalScore(name=BALANCED_ACCURACY_SCORE, value=1 / 2),
     EvalScore(name=PRECISION_SCORE, value=2 / 3),
     EvalScore(name=RECALL_SCORE, value=2 / 3),
+    EvalScore(name=UNKNOWN_RATIO, value=0.0),
+]
+
+DATASET_SCORES_WO_CONFIG = [
```
Comment (danielezhu) [inline test/unit/eval_algorithms/test_classification_accuracy.py:84]: Can you explain what this represents? I see that you replaced all instances of `DATASET_SCORES` with `DATASET_SCORES_WO_CONFIG`. Shouldn't we still have some tests that use `DATASET_SCORES`? Also, could you please come up with a case where the expected precision and recall scores are different?
